lib/hexpm_web/controllers/package_controller.ex

```diff
@@ -81,6 +81,32 @@ defmodule HexpmWeb.PackageController do
     end || not_found(conn)
   end
 
+  def audit_logs(conn, params) do
+    %{"repository" => repository, "name" => name} = params
+    organizations = Users.all_organizations(conn.assigns.current_user)
+    repositories = Map.new(organizations, &{&1.repository.name, &1.repository})
+
+    if repository = repositories[repository] do
+      package = repository && Packages.get(repository, name)
+
+      if package do
```

Comment:
Not sure whether or not to extract L84-L92 to a helper function like `access_package`, because `show/2` uses a lot of intermediate values (e.g. `repositories`)
if we use a helper function, then there's a lot values passing back-and-forth

Reply:
I think this is fine. The risk with helper functions is that we query things from the database we don't need.

Reply:
Not sure if I understand correctly, but I extracted the helper function in 9cfe301b8a6161d9d21a7575ef1e531cbdbc1b36 anyway